### PR TITLE
Add hcloud_floating_ip module

### PR DESCRIPTION
**Pull request by LKaemmerling**
##### SUMMARY

This PR aims to add a hcloud_floating_ip module

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

hcloud_ssh_key
##### ADDITIONAL INFORMATION

Follow up of #53062 and #52553.

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/hcloud/hcloud_floating_ip.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+short_description: Create and manage Floating IPs on the Hetzner Cloud
+
+version_added: "2.8"
+
+description:
+    - "Create, update and manage Floating IPss on the Hetzner Cloud."
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
```
Comment (gundalow) [inline lib/ansible/modules/cloud/hcloud/hcloud_floating_ip.py]: Case needs to match
```suggestion
    - Lukas Kaemmerling (@LKaemmerling)
```

**Comment by LKaemmerling** [comment]
@resmo i have applied some fixes from #53550 here. Should be complete now.

**Review by resmo — COMMENTED** [review]
This module doesn't seem to be idempotent. This is a known issue also seen in other modules with the same functionality getting an ip address dynamically and not yet known, luckily the library and api allows to filter by label https://github.com/hetznercloud/hcloud-python/blob/master/hcloud/floating_ips/client.py#L183.

I would suggest to change the logic by implementing label filter to gain idempotency. A similar workflow was implemented for `cs_ip_address`.

Define a label key, e.g. "name" or "id" with a unique value to get the existing floating-ip record.

**Comment by LKaemmerling** [comment]
We will search for a solution for this. We don't expect to have a solution for ansible 2.8 (because according to the [Roadmap](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_8.rst) we would have only 9 days to perform this change on our side.)

Closing for now. We will reopen a PR when we are ready for this.

**Pull request closed without merging**